Assets/Scripts/Characters/PlayerMovement.cs

```diff
@@ -1,47 +1,83 @@
 ﻿using System;
+using Mirror;
 using UnityEngine;
 using UnityEngine.InputSystem;
 using Vector2 = UnityEngine.Vector2;
 
 namespace Characters
 {
-    public class PlayerMovement : MonoBehaviour
+    public class PlayerMovement : NetworkBehaviour
     {
         [SerializeField]
         private float speed = 5f;
 
         [SerializeField]
         private LayerMask terrainLayerMask;
 
+        public static event Action<int> ServerMovementStarted;
+        public static event Action<int> ServerMovementEnded;
+        public static event Action<int> ServerPlayerJumped;
+
         private PlayerInput _playerInput;
         private Rigidbody2D _body;
         private Collider2D _collider;
         private Animator _animator;
 
         private bool _isJumping = false;
+        private bool _isMoving = false;
         private Transform _groundChecker;
 
         private static readonly int IsRunningID = Animator.StringToHash("isRunning");
         private static readonly int IsJumpingID = Animator.StringToHash("isJumping");
 
-        private void Awake()
+        public void Awake()
         {
-            _animator = GetComponentInParent<Animator>();
-            _body = GetComponentInParent<Rigidbody2D>();
-            _collider = GetComponentInParent<Collider2D>();
+            _animator = GetComponent<Animator>();
+            _body = GetComponent<Rigidbody2D>();
+            _collider = GetComponent<Collider2D>();
+            _playerInput = GetComponent<PlayerInput>();
+        }
 
-            _playerInput = GetComponentInParent<PlayerInput>();
-            _playerInput.actions["Player/Jump"].performed += HandleJump;
+        private void OnDisable()
+        {
+            _isJumping = false;
+            _animator.SetBool(IsJumpingID, _isJumping);
+            _animator.SetBool(IsRunningID, false);
         }
 
-        private void OnDestroy()
+        private void Update()
         {
-            _playerInput.actions["Player/Jump"].performed -= HandleJump;
+            if (!isLocalPlayer)
+            {
+                return;
+            }
+
+            float movement = _playerInput.actions["Player/Movement"].ReadValue<float>();
+            if (_isMoving != (Mathf.Abs(movement) > 0.01f))
+            {
+                if (Mathf.Abs(movement) > 0.01f)
+                {
+                    _isMoving = true;
+                    CmdMovementStarted();
```

Comment:
Since energy lives on the server, we can't rely directly on the `PlayerInput` anymore since it's local to the client